### ISSUE TITLE
TypeScript client types are not correct when no matching object exists

### ISSUE THREAD

**Issue by samuela** [prisma/prisma#3372]
**Describe the bug**
```typescript
const foo = await db.user({id: doesNotExist});
// foo is typed as a User, but is actually null!
```

**Expected behavior**
The type of `db.user(...)` to be `User?` reflecting the fact that it's possible that no such user exists.

**Versions (please complete the following information):**
 - OS: macOS
 - `prisma` CLI: prisma/1.19.0 (darwin-x64) node-v10.11.0
 - Prisma Server: Demo server on Prisma Cloud
 - "graphql-yoga": "^1.16.7", "prisma-client-lib": "^1.19.0"

**Additional context**
Data model:

```graphql
type User {
  id: ID! @unique
  ...
}
```

**Comment by FrankSandqvist** [comment]
This seems like a pretty major oversight to me. Either it should throw if no such object exists or it should have a nullable return type

**Comment by P4sca1** [comment]
I agree that return types should be changed to User | null. Its a common mistake to not check whether the fetched object exists and changing the typings prevents such errors.
TypeORM also returns User | null so it will be easy to adopt for people who are used to TypeORM.

**Comment by pantharshit00** [comment]
This has been fixed with prisma cli version `1.32.0-beta.1` :D

I will be in tomorrow's 1.32 release as well.
